### ISSUE TITLE
Electron intermittently freezes upon new window creation

### ISSUE THREAD

**Issue by amoshg** [electron/electron#8916]
This is running electron 1.6.2 on ubuntu debian under headless condition (xvfb).

This issue happens intermittently when we create/destroy a few windows. Here is the line of code that the program freezes at: 

```js
const nb = new electron.BrowserWindow({
                width,
                height,
                enableLargerThanScreen: true,
                backgroundColor: '#FFFFFF', // needs to be set otherwise transparent images will have black background
                show: false,
                frame: false,
                webPreferences: {
                    devTools: false,
                    webaudio: false,
                    webgl: false,
                    nodeIntegration: false,
                    preload: path.join(__dirname, 'preload.js'),
                    backgroundThrottling: false,
                },
            });
```

if I keep doing this enough times I can reproduce this consistently, this seems to happen only in our Linux environment. We are doing clean up and have checked that there are no lingering windows hanging around and that the memory usage is stable.

Upon digging a bit deeper we have found that it seems to be getting stuck at the following line in the browser-window.js 

```js
 if (process.platform !== 'darwin') {
    const menu = app.getApplicationMenu()
    if (menu) this.setMenu(menu)
  }
```

commenting out the line above seems to fix the issue, not sure what the implications are though. Any thoughts on this?

**Comment by kevinsawicki** [comment]
> commenting out the line above seems to fix the issue, not sure what the implications are though. Any thoughts on this?

Is your app setting a menu anywhere?

**Comment by amoshg** [comment]
no it's not. it's a headless app used for generating pdfs

**Comment by kiavashps** [comment]
@amoshg Just in case it helps: I was running into a similar issue on ubuntu 16.04 using Xvfb. Electron would sometimes just hang and stay open indefinitely, and sometimes would just crash immediately before it ran any of my app code.

I figured out that both cases happened because Xvfb crashed.

I figured this out by enabling logging (`export ELECTRON_ENABLE_LOGGING=true`). You might want to try that out if you haven't already
